lazyflow/slot.py

```diff
@@ -1479,15 +1479,22 @@ def _changed(self):
             # call changed callbacks
             self._sig_changed(self)
 
+    def _invokeSetupOutputs(self):
+        if self.operator.configured():
+            self.operator._setupOutputs()
+
     def _configureOperator(self, slot, oldSize=0, newSize=0, notify=True):
         """Call setupOutputs of Operator if all slots of the operator
         are connected and configured.
 
         """
         if self.operator is not None:
             # check whether all slots are connected and notify operator
-            if self.operator.configured():
-                self.operator._setupOutputs()
+            transaction = getattr(self.operator, "_current_transaction", None)
+            if transaction:
+                transaction.on_exit(self._invokeSetupOutputs)
```

Comment:
I think that we have some code like this in some of the workflows:
```
op1 = SomeOperator(...)
op2 = SomeOtherOperator(...)
op2.SomeOtherInput.connect(op1.SomeInput)
```
that is, the upstream slot doesn't have to be an OutputSlot. If you, then, do this:`
```
with op1.transaction:
    op1.SomeInput.setValue(123)
```
`op2` there would detect the change in `SomeOtherInput` and call its own `setupOutpus`. You might want to double check my reasoning here, though =)

Reply:
Updated, see [this test case](https://github.com/ilastik/lazyflow/pull/303/files#diff-5eb200b7b0b313e34a894843a84f0a14R734)